src/corehost/cli/test/nativehost/comhost_test.cpp

```diff
@@ -0,0 +1,137 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "comhost_test.h"
+#include <error_codes.h>
+#include <iostream>
+#include <future>
+#include <pal.h>
+
+namespace
+{
+    using DllGetClassObject_fn = HRESULT(STDMETHODCALLTYPE*)(REFCLSID rclsid, REFIID riid, LPVOID *ppv);
+
+    class comhost_exports
+    {
+    public:
+        comhost_exports(const pal::string_t &comhost_path)
+        {
+            if (!pal::load_library(&comhost_path, &_dll))
+            {
+                std::cout << "Load library of comhost failed" << std::endl;
+                return;
+            }
+
+            get_class_obj_fn = (DllGetClassObject_fn)pal::get_symbol(_dll, "DllGetClassObject");
+            if (get_class_obj_fn == nullptr)
+            {
+                std::cout << "Failed to get DllGetClassObject export from comhost" << std::endl;
+                return;
+            }
+        }
+
+        ~comhost_exports()
+        {
+            pal::unload_library(_dll);
+        }
+
+        DllGetClassObject_fn get_class_obj_fn;
+        bool is_valid() { return _dll != nullptr && get_class_obj_fn != nullptr; }
```

Comment:
Instead of this, you could throw an exception in the constructor. I don't see a strong argument for the `is_valid` semantic.